packages/flutter_test/lib/src/window.dart

```diff
@@ -616,62 +625,120 @@ class FakeAccessibilityFeatures implements ui.AccessibilityFeatures {
+class FakeViewPadding implements ViewPadding {
+  /// Instantiates a new [FakeViewPadding] object for faking insets and padding
+  /// during tests.
+  const FakeViewPadding({
+    this.left = 0.0,
+    this.top = 0.0,
+    this.right = 0.0,
+    this.bottom = 0.0,
+  });
+
+  FakeViewPadding._wrap(ViewPadding base) :
+    left = base.left,
+    top = base.top,
+    right = base.right,
+    bottom = base.bottom;
+
+  @override
+  final double left;
+
+  @override
+  final double top;
+
+  @override
+  final double right;
+
+  @override
+  final double bottom;
+}
+
 /// [PlatformDispatcher] that wraps another [PlatformDispatcher] and
 /// allows faking of some properties for testing purposes.
 ///
 /// See also:
 ///
+///   * [TestFlutterView], which wraps a [FlutterView] for testing and
+///     mocking purposes.
 ///   * [TestWindow], which wraps a [SingletonFlutterWindow] for
 ///     testing and mocking purposes.
-class TestPlatformDispatcher implements ui.PlatformDispatcher {
+class TestPlatformDispatcher implements PlatformDispatcher {
   /// Constructs a [TestPlatformDispatcher] that defers all behavior to the given
-  /// [dart:ui.PlatformDispatcher] unless explicitly overridden for test purposes.
+  /// [PlatformDispatcher] unless explicitly overridden for test purposes.
   TestPlatformDispatcher({
-    required ui.PlatformDispatcher platformDispatcher,
-  }) : _platformDispatcher = platformDispatcher;
+    required PlatformDispatcher platformDispatcher,
+  }) : _platformDispatcher = platformDispatcher {
+    _updateViews();
+  }
 
-  /// The [dart:ui.PlatformDispatcher] that is wrapped by this [TestPlatformDispatcher].
-  final ui.PlatformDispatcher _platformDispatcher;
+  /// The [PlatformDispatcher] that is wrapped by this [TestPlatformDispatcher].
+  final PlatformDispatcher _platformDispatcher;
 
   @override
-  ui.VoidCallback? get onMetricsChanged => _platformDispatcher.onMetricsChanged;
+  TestFlutterView? get implicitView {
+    return _platformDispatcher.implicitView != null
+      ? _testViews[_platformDispatcher.implicitView!.viewId]!
+      : null;
+  }
+
+  final Map<Object, TestFlutterView> _testViews = <Object, TestFlutterView>{};
+
+  @override
+  VoidCallback? get onMetricsChanged => () {
+    _updateViews();
+    _platformDispatcher.onMetricsChanged?.call();
+  };
+
   @override
-  set onMetricsChanged(ui.VoidCallback? callback) {
+  set onMetricsChanged(VoidCallback? callback) {
     _platformDispatcher.onMetricsChanged = callback;
   }
 
   @override
-  ui.Locale get locale => _localeTestValue ?? _platformDispatcher.locale;
-  ui.Locale? _localeTestValue;
+  Locale get locale => _localeTestValue ?? _platformDispatcher.locale;
+  Locale? _localeTestValue;
   /// Hides the real locale and reports the given [localeTestValue] instead.
-  set localeTestValue(ui.Locale localeTestValue) { // ignore: avoid_setters_without_getters
+  set localeTestValue(Locale localeTestValue) { // ignore: avoid_setters_without_getters
```

Comment:
We should probably file an issue (and maybe add a TODO here) to move the TestPlatformDispatcher over to the same API used by TestFlutterView?

Reply:
Yeah, it's on my list of issues to file today...

Reply:
Filed! #121901